tests/Integration/Service/AntiSpamServiceIntegrationTest.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+/**
+ * @copyright 2017 Christoph Wurst <christoph@winzerhof-wurst.at>
+ *
+ * @author 2017 Christoph Wurst <christoph@winzerhof-wurst.at>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Mail\Tests\Integration\Service;
+
+use Horde_Imap_Client;
+use OC;
+use OCA\Mail\Account;
+use ChristophWurst\Nextcloud\Testing\TestCase;
+use OCA\Mail\Contracts\IMailManager;
+use OCA\Mail\Contracts\IMailTransmission;
+use OCA\Mail\Db\MessageMapper;
+use OCA\Mail\IMAP\MessageMapper as ImapMessageMapper;
+use OCA\Mail\Service\AntiSpamService;
+use OCA\Mail\Service\Sync\SyncService;
+use OCA\Mail\Tests\Integration\Framework\ImapTest;
+use OCA\Mail\Tests\Integration\Framework\ImapTestAccount;
+use OCP\IConfig;
+
+class AntiSpamServiceIntegrationTest extends TestCase {
+	use ImapTest,
+		ImapTestAccount;
+
+	/** @var AntiSpamService */
+	private $service;
+
+	public function setUp():void {
+		parent::setUp();
+		$this->service = new AntiSpamService(
```

Comment:
you could also fetch the whole `AntiSpamService` from the container and thereby indirectly test if the container can build the service. This has the nice side effect that new arguments to the constructor don't have to be added here.